### DISCUSSION TITLE
`vim.ui_attach` callback: is it correct to get `msg_showmode` events with `noshowmode`?

### DISCUSSION THREAD

**Discussion by przepompownia** [neovim/neovim#31629] [Q&A]
I'm not sure if this is intentional or a bug (or they are unrelated things). 

For comparison `msg_showcmd` events trigger the callback iff `showcmd` is set.

**Comment by luukvbaal** [comment]
Current logic indeed still emits redundant showmode clear events (empty content `{}`), seems like a bug/oversight to me. You shouldn't be getting events with actual content? Fixed in #32167.

**Reply by przepompownia** [reply]
> You shouldn't be getting events with actual content?

Probably I don't understand this question. Both with and without `showmode` I get messages with empty content. While I don't need any of them, I tried to avoid triggering the callback by those events in a way similar to `noshowcmd` case.
